src/Query/Builder.php

```diff
@@ -1191,10 +1193,41 @@ protected function compileWhereDate(array $where): array
     {
         extract($where);
 
-        $where['operator'] = $operator;
-        $where['value'] = $value;
+        $startOfDay = new UTCDateTime(Carbon::parse($value)->startOfDay());
+        $endOfDay = new UTCDateTime(Carbon::parse($value)->endOfDay());
 
-        return $this->compileWhereBasic($where);
+        return match($operator) {
+            'eq', '=' => [
+                $column => [
+                    '$gte' => $startOfDay,
+                    '$lte' => $endOfDay,
+                ],
+            ],
+            'ne' => [
+                '$or' => [
```

Comment:
I tested this alternative, but got segfaults.
```php
'ne' => ['$not' => [
    $column => [
        '$gte' => $startOfDay,
        '$lte' => $endOfDay,
    ],
]],
```

Reply:
> got segfaults

Segfaults in `mongod`? I can't imagine how this would trigger a segfault in PHP.